muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/ReferenceMatcher.java

```diff
@@ -294,13 +294,19 @@ private static MethodDescription.InDefinedShape findMethod(
       return null;
     }
 
+    TypeDescription superType = null;
     if (typeOnClasspath.getSuperClass() != null) {
-      MethodDescription.InDefinedShape methodOnSupertype =
-          findMethod(methodRef, typeOnClasspath.getSuperClass().asErasure());
+      superType = typeOnClasspath.getSuperClass().asErasure();
+    } else if (!"java.lang.Object".equals(typeOnClasspath.getName())) {
+      superType = TypeDescription.ForLoadedType.of(Object.class);
```

Comment:
Just curious, what was this needed for?

Reply:
If I remember correctly muzzle failed when one of the `Object` methods (`hashCode`, `equals` ... don't remember which) was called on an interface type.